### PR TITLE
Vore tweaks

### PR DESCRIPTION
**Pull request by Poojawa**
Work to ensure digestion functions as intented

Devourable state can be triggered client side. Will prevent and vore actions upon your character.

Shouldn't die to pressure loss or cold/heat whilst inside someone.

Can't see shit inside someone anymore either, counts as restrained for movement as well, so no internal shaking/healing/bullshit.

### CODE REVIEW

File: code/citadel/vore/resizing/resize_vr.dm [hunk 1]
```diff
@@ -142,54 +143,55 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
...
-//			var/mob/living/carbon/human/M = src
-//			if(istype(M) && istype(M.tail_style, /datum/sprite_accessory/tail/taur/naga))
-//				src << "You carefully squish [tmob] under your tail!"
-//				tmob << "[src] pins you under their tail!"
-//			else
+/*
+			var/mob/living/carbon/human/H = src
+			if(istype(H) && istype(H.tail_style, /datum/sprite_accessory/tail/taur/naga))
+				src << "You carefully squish [tmob] under your tail!"
+				tmob << "[src] pins you under their tail!"
```
Comment (ghost) [inline code/citadel/vore/resizing/resize_vr.dm:154]: :+1:

**Comment by Poojawa** [comment]
Probably null'd when I get #58 working, but saving doesn't work, and selecting any other belly only works on the panel, it doesn't change the target. You can transfer between however.

**Comment by TalkingCactus** [comment]
Is this abandoned?

**Comment by ghost** [comment]
probably

**Comment by Poojawa** [comment]
Not abandoned, just haven't had time or inspiration on how to resolve the only real hold up.

Namely, ensuring that the correct gut is selected and used for the vore action rather than always being default.

**Comment by TalkingCactus** [comment]
Point me at the problem next chance you get and I'll take a crack at it.

File: code/citadel/vore/eating/belly_vr.dm [hunk 2]
```diff
@@ -138,12 +138,17 @@
...
 /datum/belly/proc/nom_mob(var/mob/prey, var/mob/user)
-//	if (prey.buckled)
-//		prey.buckled.unbuckle_mob()
+//	if (prey.anchored)
+//		prey.anchored.unbuckle_mob()
 
 // Super super messy. prey.forceMove.owner doesn't work if there's no prey.
 	prey.loc = owner
-	internal_contents |= prey
+
```
Comment (Poojawa) [inline code/citadel/vore/eating/belly_vr.dm]: This check needs to be applied at the same check to see if someone can be taken.

File: code/citadel/vore/eating/vore_vr.dm [hunk 3]
```diff
@@ -67,6 +29,16 @@ V::::::V           V::::::VO:::::::OOO:::::::ORR:::::R     R:::::REE::::::EEEEEE
...
+	set name = "Choose Belly"
+	set category = "Vore"
+
+	vore_selected = input("Choose Belly") in vore_organs
+	src << "<span class='notice'>[vore_selected] selected.</span>"
+
+//
 //	Belly searching for simplifying other procs
 //
 /proc/check_belly(atom/movable/A)
```
Comment (Poojawa) [inline code/citadel/vore/eating/vore_vr.dm:44]: This is the proc that is supposed to be triggered.

File: code/citadel/vore/eating/vorepanel_vr.dm [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 	var/dat = picker_holder.gen_vui(src)
 
-	picker_holder.popup = new(src, "insidePanel","Inside!", 400, 600, picker_holder)
+	picker_holder.popup = new(src, "insidePanel","Vore Panel", 400, 600, picker_holder)
```
Comment (Poojawa) [inline code/citadel/vore/eating/vorepanel_vr.dm:15]: Line 90 for belly selection via vore panel.

File: code/citadel/vore/eating/belly_vr.dm (hunk 2 above)
Comment (TalkingCactus) [reply code/citadel/vore/eating/belly_vr.dm]: nom_mob() or check_belly()?

Comment (Poojawa) [reply code/citadel/vore/eating/belly_vr.dm]: check_belly()

the action needs to realize that the user has selected something else in the vore UI menu when it goes to carry through the nom_mob().

As the guts all actually exist there, it's just a matter of getting the target selection right again. Grab code is fucking awful.

Comment (TalkingCactus) [reply code/citadel/vore/eating/belly_vr.dm]: So where are the bellies located code wise.

File: code/citadel/vore/eating/vore_vr.dm (hunk 3 above)
Comment (ghost) [reply code/citadel/vore/eating/vore_vr.dm:44]: `return vore_selected` ?

**Comment by LetterJay** [comment]
line 145 in belly_vr.dm: change prey.loc = owner to prey.loc = user

**Comment by Poojawa** [comment]
Hooray. Working. Also just so happened to preserve flavor text to print in examine.  How about that.

**Comment by Poojawa** [comment]
well, working as it compiles with modernization code.

**Comment by ghost** [comment]
congratulations

**Comment by Poojawa** [comment]
And for my next trick, trying to get the release working.

In theory, it'd be prey.forceMove(owner.loc) ((or something close to that))
and owner.internal_contents -= prey

Because that's basically what the nom mob proc is.

but if you switch the internal contents to remove the prey first, then move them to the location... internal_contents becomes an undefined verb. ????????

Fuck it, going to bed

**Comment by Poojawa** [comment]
Closed until other coders stop being lazy and update their spaghetti code.

**Pull request closed without merging**